mkosi/__init__.py

```diff
@@ -6610,6 +6662,11 @@ def setup_ssh(
     if args.ssh_key:
         f = open(args.ssh_key, mode="r", encoding="utf-8")
         copy_file(f"{args.ssh_key}.pub", authorized_keys)
+    elif args.ssh_agent is not False:
+        env = {"SSH_AUTH_SOCK": args.ssh_agent}
+        result = run(["ssh-add", "-L"], env=env, text=True, stdout=subprocess.PIPE)
+        authorized_keys.write_text(result.stdout)
+        f = cast(TextIO, result.stdout)
```

Comment:
not sure this is legal/sensible...

Reply:
Yeah this seems a bit iffy, can't you just set `f` to `None`, we return an `Optional` so that should just work I think?

Reply:
ok, done

Reply:
nope:

```
mkosi/__init__.py:6669:13: error: Incompatible types in assignment (expression
has type "None", variable has type "TextIO")  [assignment]
            f = None
```

Reply:
The error is because `f` is declared as `TextIO`  a little earlier in the function. Just change that to `Optional[TextIO]` and it should work

Reply:
done - one day I will understand all this typing stuff. One day...